habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDepsMojo.java

```diff
@@ -41,93 +38,57 @@ public class ContainerizeDepsMojo extends AbstractHabushuMojo {
-
-    /**
-     * Working directory relative to the basedir - typically working directory and basedir are synonymous.
-     */
-    @Parameter(readonly = true)
-    protected String workingDirectoryRelativeToBasedir;
-
-    /**
-     * Expected subdirectory of the monorepo dependency's basedir
-     * in which Poetry places generated source and wheel archive distributions.
-     */
-    @Parameter(defaultValue = "/dist", readonly = true)
-    protected String distDirectoryRelativeToBasedir;
-
-    /**
-     * Expected subdirectory of the monorepo dependency's basedir
-     * in which Maven places build-time artifacts. Should NOT include dist items.
-     */
-    @Parameter(defaultValue = "/target", readonly = true)
-    protected String targetDirectoryRelativeToBasedir;
+    @Parameter(defaultValue = "true", property = "habushu.updateDockerfile")
+    protected boolean updateDockerfile;
 
     /**
-     * Location of where containerization files will be placed.
+     * Dockerfile to be updated with containerization logic. Must be set if `updateDockerfile` is true.
      */
-    @Parameter(defaultValue = "${project.build.directory}/containerize-support", readonly = true)
-    protected String containerizeSupportDirectory;
+    @Parameter(property = "habushu.dockerfile")
+    protected File dockerfile;
 
     /**
-     * Upstream directory that houses all necessary monorepo dependencies.
-     * Monorepo dependency source files will be copied from here.
+     * The directory that will serve as the context for the Docker build. This directory must contain the `stagingDirectory`.
+     * Defaults to the project's base directory.
      */
-    @Parameter(readonly = true)
-    protected File anchorSourceDirectory;
+    @Parameter(defaultValue = "${project.basedir}", property = "habushu.dockerContext")
+    protected File dockerContext;
 
     /**
-     * Update dockerfile
+     * The user to set as the owner of the virtual env. This is useful when the Docker build is run as a non-root user.
+     * Set to an empty string to disable.
      */
-    @Parameter(defaultValue = "true", property = "habushu.updateDockerfile")
-    protected boolean updateDockerfile;
+    @Parameter(defaultValue = "1001", property = "habushu.dockerUser")
+    protected String dockerUser;
 
     /**
-     * Dockerfile to be updated with the stage content
+     * The base image to use for the Dockerfile. The base image will be used both to bundle the virtual environment for
+     * the target project and to run the final container, as the venv must be built on the same platform as the final
+     * runtime. The base image must have the correct Python version resolvable via the PATH.
      */
-    @Parameter(property = "habushu.dockerfile")
-    protected File dockerfile;
-
-    private Path anchorOutputDirectory;
+    @Parameter(defaultValue = "python:3.11-slim", property = "habushu.dockerBase")
+    protected String dockerBase;
```

Comment:
Should we add `dockerUser` and `dockerBase` to the README?

Reply:
We've talked a bit about trying to move away from our current exhaustive pattern of documentation, as it's become so unwieldy that its usefulness is questionable.  But maybe it makes sense to at least touch on `dockerBase` in the main documentation for the goal/feature? I'll see if it's a natural fit there while still being succinct.